keps/sig-network/4974-deprecate-endpoints/README.md

```diff
@@ -0,0 +1,641 @@
+  associated tests skipped, and confirm that it passes.
+
+#### Deprecation - Stage 3
+
+Stage 3 will not happen until the relevant SIGs feel that enough time
+has passed since the initial deprecation announcement.
+
+- The tests depending on the Endpoints and EndpointSlice Mirroring
+  controllers are demoted from conformance, and there is additional
+  communication (e.g. blog post) about this.
+
+- The documentation is updated to explain how to disable the Endpoints
+  and EndpointSlice Mirroring controllers, but warns that third-party
+  components may still depend on them.
+
+#### Deprecation - Stage 4
+
+Stage 4 will not happen until the relevant SIGs feel reasonably
+confident that the Kubernetes ecosystem has mostly moved away from
+Endpoints.
+
+- The documentation becomes more bullish on the idea of disabling the
+  controllers, implying that it's a reasonable default.
+
+### Upgrade / Downgrade Strategy
+
+The KEP does not propose any automatic change to behavior; behavior
+would only be changed when the administrator chose to disable the
+controllers, which could happen at any time.
+
+### Version Skew Strategy
+
+The only non-opt-in behavioral change is fixing the apiserver
+aggregation and proxying APIs to use EndpointSlices rather than
+Endpoints internally, which does not present any skew issues (since
+EndpointSlices have already existed for a long time at this point).
+
+(When the conformance criteria change to allow disabling the Endpoints
+controller, this would only apply to clusters that are _fully_ at the
+new version, to avoid skew issues. More specifically: it is not
+conforming to disable the Endpoints controller if you still have any
+apiservers that use Endpoints for apiserver aggregation and proxying.)
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+The KEP does not define a new feature, it just proposes that we
+document that users are allowed to disable the Endpoints and
+EndpointSlice Mirroring controllers.
+
+###### Does enabling the feature change any default behavior?
+
+The KEP does not define a new feature that can be enabled.
+
+(Obviously disabling the endpoints controllers changes default
+behavior, but this would be because the administrator chose to do
+that, not something that would happen automatically.)
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The KEP does not define a new feature that can be enabled/disabled.
+
+(If an administrator chooses to disable the endpoints controllers, and
+then decides this was a bad idea, they can re-enable them, and even if
+something had previously deleted all of the autogenerated Endpoints
+objects, re-enabling the endpoints controller would regenerate them.)
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A
```

Comment:
Not a blocker now, but I think that before `Stage 3`, we actually should test endpoints-controller disablement -> enablement -> disablement (again).

Can you please mention that such test will be run before stage 3 (I don't want to block you by bothering you about full description now).

Reply:
Sure. Is there some particular failure mode you're worried about? (Is this more about the stale Endpoints cleanup than about the Endpoints controller itself?)